libvirt/qemu_agent.go

```diff
@@ -24,75 +26,98 @@ type QemuAgentInterfaceIpAddress struct {
 	Prefix  uint   `json:"prefix"`
 }
 
+type libvirtIfacesLst []libvirt.DomainInterface
+
 // Retrieve all the interfaces attached to a domain and their addresses. Only
 // the interfaces with at least an IP address are returned.
 // When wait4ipv4 is turned on the code will not report interfaces that don't
 // have a ipv4 address set. This is useful when a domain gets the ipv6 address
 // before the ipv4 one.
-func getDomainInterfacesViaQemuAgent(domain LibVirtDomain, wait4ipv4 bool) []libvirt.DomainInterface {
-	log.Print("[DEBUG] get network interfaces using qemu agent")
-
-	var interfaces []libvirt.DomainInterface
-
-	result, err := domain.QemuAgentCommand(
-		"{\"execute\":\"guest-network-get-interfaces\"}",
-		libvirt.DOMAIN_QEMU_AGENT_COMMAND_DEFAULT,
-		0)
-	if err != nil {
-		return interfaces
-	}
+func getDomainInterfacesViaQemuAgent(domain LibVirtDomain, wait4ipv4 bool) libvirtIfacesLst {
 
-	log.Printf("[DEBUG] qemu-agent response: %s", result)
+	qemuAgentInterfacesRefreshFunc := func() resource.StateRefreshFunc {
```

Comment:
Since this function is quite long, we should declare it outside of `getDomainInterfacesViaQemuAgent`.